### DISCUSSION TITLE
How to train a classifier for a situation that has imbalanced data using the support provided by MONAI?

### DISCUSSION THREAD

**Discussion by DSRajesh** [Project-MONAI/MONAI#7111] [Q&A]
Hello

We are building a 3D volume classifier  for classifying  an object image into one of three classes. We expect the classifier to output 3 categorical outputs each one corresponding to one of the 3 classes.    The dataset we are using is imbalanced such that object images (3D) corresponding to the first class are plenty in number when compared to the number of  the other two classes i.e the numbers of the images of 3 classes  is approximately 900, 57, 38 respectively.    We have tried with DenseNet121 network and some simple augmentation techniques provided on MONAI website, like random rotation and scaling, but not with much success.  If you could please suggest some more utilities of MONAI to handle data imbalance (like stratified sampling, positive-negative sampling (hard_negative_sampler)) such a situation it will be helpful.  Thanking you

Rajesh

**Comment by KumoLiu** [comment]
Hi @DSRajesh, thanks for your interest here.
You can try these two transforms according to your situation.
https://github.com/Project-MONAI/MONAI/blob/4743945a68952d05c8a97b82b1bb75a3ad5dbf71/monai/transforms/croppad/array.py#L1049
https://github.com/Project-MONAI/MONAI/blob/4743945a68952d05c8a97b82b1bb75a3ad5dbf71/monai/transforms/croppad/array.py#L1228

Hope it help, thanks!
